lib/viewmodels/map_view_model.dart

```diff
@@ -0,0 +1,33 @@
+import "dart:async";
+import "package:google_maps_flutter/google_maps_flutter.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+
+class MapInfo {
+  MapInfo({
+    required this.currentLocation,
+  });
+  final LatLng currentLocation;
+}
+
+// TODO: For now, this is code with mock data.
+// This will just be replaced when we implement a real view model anyway.
+class MapViewModel extends AsyncNotifier<MapInfo> {
+  @override
+  Future<MapInfo> build() async {
+    return MapInfo(
+      currentLocation: const LatLng(46.519653, 6.632273),
+    );
+  }
+
+  /// Refresh the mapInfo.
+  /// This will put the state of the viewmodel to loading
+  /// and update the state accordingly
+  Future<void> refresh() async {
+    state = const AsyncValue.loading();
+    state = await AsyncValue.guard(() => build());
```

Comment:
While checking the coverage, I noticed that this was not covered by the tests.
This makes perfect sense given that the view model is only a place holder for the future real implementation.
Thus, I was wondering if we shouldn't keep the refresh method empty for this PR and then it will be implemented during the PR for the map view model.
I wonder what are your thoughts on this ?

Reply:
Even if it's not tested yet, I think that the logic of the button should already by implemented, as one of the acceptance criteria implies that the user can retry to load the screen when an error occurs. On top of that, those specific lines will be covered in the next PR's, with the implementation of the real user location arriving soon.

Reply:
This seems good to me then :+1: